src/session/session.c

```diff
@@ -905,6 +905,12 @@ pass_to_child (int signo)
     kill (child, signo);
 }
 
+static void
+on_authorize_timeout (int signo)
+{
+  errx (EX, "timed out waiting for authorize response");
```

Comment:
`errx()` calls into the `printf()` machinery which isn't signal-safe due to the stdio lock.

Reply:
Ah, thanks -- FTR, not installing a signal handler at all would work as well, but I thought a nicer error message would be helpful. According to https://man7.org/linux/man-pages/man7/signal-safety.7.html, write(2) and _exit() are signal safe, so  I changed it to that. Tested again with

```diff
--- src/session/session.c
+++ src/session/session.c
@@ -962,7 +962,7 @@ main (int argc,
   /* And get back the authorization response; this should not take forever, time-bound it to guard
    * against hacking/PID recycling attempts */
   sigaction (SIGALRM, &(struct sigaction) { .sa_handler = on_authorize_timeout, .sa_flags = 0 }, NULL);
-  alarm (60);  /* like cockpit_ws_auth_response_timeout on the ws side */
+  alarm (1);  /* like cockpit_ws_auth_response_timeout on the ws side */
   char *authorization = read_authorize_response ("authorization");
   alarm (0);
   char *rhost = get_authorize_key (authorization, "remote-peer", false) ?: "";
diff --git src/ws/cockpitauth.c src/ws/cockpitauth.c
index 48d3f42f9..c50c2390b 100644
--- src/ws/cockpitauth.c
+++ src/ws/cockpitauth.c
@@ -621,6 +621,7 @@ reply_authorize_challenge (CockpitSession *session)
   if (cockpit_authorize_type (session->authorization, &authorization_type) &&
       (g_str_equal (authorize_type, "*") || g_str_equal (authorize_type, authorization_type)))
     {
+      sleep(5);
       const gchar *remote_peer = cockpit_creds_get_rhost (cockpit_web_service_get_creds (session->service));
       send_authorize_reply (session->transport, cookie, session->authorization, remote_peer);
       cockpit_memory_clear (session->authorization, -1);

```